### ISSUE TITLE
networking.tempAddresses = "disabled" and networking.networkmanager

### ISSUE THREAD

**Issue by benneti** [NixOS/nixpkgs#139306]
### Describe the bug
When setting `networking.tempAddresses = "disabled"` I would expect this to be respected even when enabling networkmanager.
But I think the 
```
[connection]
ipv6.ip6-privacy=2
```
in the default networkmanager configuration overrides this

### Steps To Reproduce
1. networking.tempAddresses = "disabled";
2. networking.networkmanager.enable = "true";
3. check ip

### Expected behavior
respect global setting

### Additional context
manually adding 
```
[connection]
ipv6.ip6-privacy=0
```
to networking.networkmanager.extraConfig works
### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: networking.networkmanager.enable networking.tempAddresses
```

**Comment by Artturin** [comment]
Please test https://github.com/NixOS/nixpkgs/pull/139337
https://nixos.wiki/wiki/Nixpkgs/Reviewing_changes#Modules

**Comment by benneti** [comment]
Can you tell the bot to create a backports PR, I think this one does not work with nixos-21.05:
 error: The option `hardware.wirelessRegulatoryDatabase' does not exist. Definition values:
```
        - In `/nix/store/nn44ax0qwddg72ykzh9vnqmwhn1fn79k-source/nixos/modules/services/networking/networkmanager.nix':
            {
              _type = "if";
              condition = true;
              content = true;
            }
```
If there is no other way I can also check against unstable.

**Comment by Artturin** [comment]
Replace this module too https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/hardware/all-firmware.nix

**Comment by benneti** [comment]
ok, also needs libreelec-dvb-firmware from master but then it works.
Ip adresses are preserved over a reboot, too. So I think it does what it should and NetworkManager now respects the dbus (or however to call it) settting!

Thanks for the quick reaction!

**Comment by Artturin** [comment]
Does setting the temp address option to true still work

**Comment by benneti** [comment]
setting it to `"default"` seems to not change the ipv6 address, so I guess it does not...

I think the problem is that the udev rule is not applied for some reason. 
/etc/udev/rules.d/98-ipv6-privacy-extensions.rules
```
# enable and prefer IPv6 privacy addresses by default
ACTION=="add", SUBSYSTEM=="net", RUN+="/nix/store/wv35g5lff84rray15zlzarcqi9fxzz84-bash-4.4-p23/bin/sh -c 'echo 2 > /proc/sys/net/ipv6/conf/%k/use_tempaddr'"
```
which is what I would expect for this setting but
cat /proc/sys/net/ipv6/conf/default/use_tempaddr
`0`
(same for the interface I use)
Lastly the contents of are what I would expect
```
[main]
dhcp=internal
dns=default
firewall-backend=iptables
plugins=keyfile
rc-manager=resolvconf

[keyfile]


[logging]
audit=no
level=WARN

[connection]
ethernet.cloned-mac-address=preserve
wifi.cloned-mac-address=preserve

[device]
wifi.backend=wpa_supplicant
wifi.scan-rand-mac-address=yes
```

**Comment by benneti** [comment]
Manually writing the 2 in the files applies results in the correct setting, so it probably is due to the udev rule

**Comment by Artturin** [comment]
`journalctl | grep -i tempaddr` any errors?

**Comment by benneti** [comment]
oh yeah, lots of 
```
 Sep 25 17:39:30 HOSTNAME sudo[3524]:  USERNAME : TTY=pts/0 ; PWD=/home/USERNAME ; USER=root ; COMMAND=/n
 ix/store/wv35g5lff84rray15zlzarcqi9fxzz84-bash-4.4-p23/bin/sh -c echo 2 > /proc/sys/net/ipv6/conf/INTERFACE/use_tempaddr
```
for INTERFACES I actually use and do not use

**Comment by benneti** [comment]
sorry closed on accident

**Comment by Artturin** [comment]
i have not applied the changes but i get ```systemd-udevd[228394]: usb0: Process '/nix/store/1flh34xxg9q3fpc88xyp2qynxpkfg8py-bash-4.4-p23/bin/sh -c 'echo 2 > /proc/sys/net/ipv6/conf/usb0/use_tempaddr'' failed with exit code 1.```

which is because usb0 gets renamed to enp42s0f3u2
`udevadm monitor --property` + replug

```
UDEV  [13018.425478] move     /devices/pci0000:00/0000:00:01.2/0000:20:00.0/0000:21:08.0/0000:2a:00.3/usb3/3-2/3-2:1.0/net/enp42s0f3u2 (net)
ACTION=move
DEVPATH=/devices/pci0000:00/0000:00:01.2/0000:20:00.0/0000:21:08.0/0000:2a:00.3/usb3/3-2/3-2:1.0/net/enp42s0f3u2
SUBSYSTEM=net
DEVPATH_OLD=/devices/pci0000:00/0000:00:01.2/0000:20:00.0/0000:21:08.0/0000:2a:00.3/usb3/3-2/3-2:1.0/net/usb0
INTERFACE=enp42s0f3u2
```

so maybe the solution is to run the command and ACTION=move aswell

**Comment by benneti** [comment]
fixed in unstable (and probably 21.11)
